### PR TITLE
Add MACE-OFF23 models

### PR DESCRIPTION
**Pull request by davkovacs**
This PR implements the interface to the MACE-OFF23 models.

### CODE REVIEW

**Comment by ilyes319** [comment]
@davkovacs I think we should also add a paragraph in the readme about the models

File: mace/calculators/foundations_models.py [hunk 1]
```diff
@@ -121,6 +121,72 @@ def mace_mp(
     return calc
 
 
+def mace_off(
+    model: Union[str, Path] = None,
+    device: str = "",
+    default_dtype: str = "float32",
```
Comment (jthorton) [inline mace/calculators/foundations_models.py:127]: The default dtype does not match the doc string.

Comment (davkovacs) [reply mace/calculators/foundations_models.py:127]: good catch, I am updating it.

**Comment by davkovacs** [comment]
Ready to be merged.

**Comment by jthorton** [comment]
Might be good to add a test as well to make sure nothing gets broken in the future?

**Comment by ilyes319** [comment]
You can use the same test as the mace_mp

**Comment by davkovacs** [comment]
Can you test the tests please! @jthorton

**Comment by ilyes319** [comment]
ready to merge?

File: mace/calculators/foundations_models.py [hunk 2]
```diff
@@ -121,6 +121,79 @@ def mace_mp(
...
+            # download and save to disk
+            print(f"Downloading MACE model from {checkpoint_url!r}")
+            print(
+                f"By downloading the model you accept the ASL license, see https://github.com/gabor1/ASL"
+            )
+            urllib.request.urlretrieve(checkpoint_url, cached_model_path)
+            print(f"Cached MACE model to {cached_model_path}")
+        model = cached_model_path
+        msg = f"Using MACE-OFF23 MODEL for MACECalculator with {model}"
+        print(msg)
```
Comment (jthorton) [inline mace/calculators/foundations_models.py:174]: Looks like we missed this print which should only trigger if we don't just want to return the model.

**Pull request merged**